core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala

```diff
@@ -46,8 +46,10 @@ object BrokerMetadataPublisher {
   def getTopicDelta(topicName: String,
                     newImage: MetadataImage,
                     delta: MetadataDelta): Option[TopicDelta] = {
-    Option(newImage.topics().getTopic(topicName)).map {
-      topicImage => delta.topicsDelta().changedTopic(topicImage.id())
+    Option(newImage.topics().getTopic(topicName)).flatMap {
```

Comment:
Is `newImage.topics()` never null? If so, should we add null check to constructor of `MetadataImage` to make sure input arguments are NOT null.

Reply:
There's just two usages of this constructor and they cannot provide nulls, so we can probably live without the null check